spec/02-integration/05-proxy/01-router_spec.lua

```diff
@@ -499,4 +499,44 @@ describe("Router", function()
         check(i, args[3], args[4]))
     end
   end)
+
+  describe("url encoded uris", function()
+
+    setup(function()
+      helpers.dao:truncate_tables()
+      assert(helpers.dao.apis:insert {
+        name         = "localbin",
+        upstream_url = "http://localhost:9999/",
+        uris         = {
+          "/",
+        },
+        hosts        = {
+          "localbin.com",
+        },
+      })
+
+      assert(helpers.start_kong {
+        nginx_conf = "spec/fixtures/custom_nginx.template",
```

Comment:
As said before, we should not rely on such a temporary "hack". Yes, using this config file was a nice way to inspect incoming request headers, but it was *temporary*. All other tests rely on httpbin.org`. This template file for unit tests of the Nginx config compilation, and I feel like the `headers-inspect` set a precedent we should not have.

Yes, we want local httpbin testing. But not without a proper implementation, otherwise we'll end up with yet another way of testing that was supposed to be temporary, and is not scalable (we can't each and every possible endpoint we need, even only once, to that poor template that was not even intended for it originally). The local httpbin needs to be implemented properly first. Let's keep using httpbin in the meanwhile.